### PR TITLE
ADL Updates for Catalog and Upload/Download robustness

### PR DESCRIPTION
**Pull request by begoldsm**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/custom.py [hunk 1]
```diff
@@ -195,6 +195,51 @@ def update_adla_catalog_credential(client,
...
+def list_catalog_tables(client,
+                        account_name,
+                        database_name,
+                        schema_name=None):
+    if not schema_name:
+        return client.list_tables_by_database(account_name, database_name)
+
+    return client.list_tables(account_name, database_name, schema_name)
+
+def list_catalog_views(client,
```
Comment (troydai) [inline src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/custom.py]: At this point, the CI build hasn't finished yet, but I guess the code style check will fail. Please run `check_style --module dla --pep8 --pylint` to detect all code styles issue.

File: src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/custom.py [hunk 2]
```diff
@@ -195,6 +195,51 @@ def update_adla_catalog_credential(client,
     client.update_credential(account_name, database_name, credential_name, update_params)
 
 
+# customizations for listing catalog items that support multiple ancestor levels of listing.
```
Comment (troydai) [inline src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/custom.py]: At this point, the CI build hasn't finished yet, but I guess the code style check will fail. Please run `check_style --module dla --pep8 --pylint` to detect all code styles issue.

**Comment by begoldsm** [comment]
@troydai thanks for the quick check! I just ran and fixed pylint and pep8 issues! Is there anything else that I should run locally as well to proactively ensure a successful CI?

**Comment by troydai** [comment]
@begoldsm run `check_style --ci` will do universal style check. And run `run_tests --module <module>` will run the tests for your command module (`run_tests --module dla` in your case).

**Pull request merged**